extensions/panache/mongodb-panache-common/runtime/src/main/java/io/quarkus/mongodb/panache/common/reactive/runtime/ReactiveMongoOperations.java

```diff
@@ -208,9 +211,12 @@ public Uni<Void> persistOrUpdate(Stream<?> entities) {
     public Uni<Void> delete(Object entity) {
         ReactiveMongoCollection collection = mongoCollection(entity);
         BsonDocument document = getBsonDocument(collection, entity);
-        BsonValue id = document.get(ID);
-        BsonDocument query = new BsonDocument().append(ID, id);
-        return collection.deleteOne(query).onItem().ignore().andContinueWithNull();
+        VersionHandler versionHandler = VersionHandler.handle(entity, document);
+
+        return collection.deleteOne(versionHandler.query).onItem()
+                .invoke(deleteResult -> versionHandler
+                        .checkHasNotAffectedResults(((DeleteResult) deleteResult).getDeletedCount()))
+                .onItem().ignore().andContinueWithNull();
```

Comment:
Is it really related to optimistic lock controll to verify that a delete operation really delete something? I'm not very compfortable and more important: if I want to delete something and someone else already deleted it why throw ?

Reply:
@loicmathieu since we always return a `Uni<Void>`, we can ignore this missing delete.
I left this there for cases where you are trying to delete something that was recently updated by another user, it is just like an alert "hey, someone updated the info you want to delete, are you sure?".

This is there because I had a case where it was useful, but I can remove, what do you think?

The point @evanchooly said about trying to delete with a non existent id makes sense for me, and I think is better to remove this check when deleting.

By return something other than void, I think void is a standard in frameworks, panache/hibernante/jpa, all works in this way, I don't think we should change this

Reply:
I've just remove this check here and for MongoOperations.